includes/class-wc-tracker.php

```diff
@@ -621,6 +624,63 @@ private static function get_order_dates() {
 
 		return array_merge( $min_max, $processing_min_max );
 	}
+
+	/**
+	 * Search a specific post for text content.
+	 *
+	 * @param integer $post_id The id of the post to search.
+	 * @param string  $text    The text to search for.
+	 * @return string 'Yes' if post contains $text (otherwise 'No').
+	 */
+	public static function post_contains_text( $post_id, $text ) {
+		global $wpdb;
+
+		// Search for the text anywhere in the post.
+		$wildcarded = "%{$text}%";
+
+		$result = $wpdb->get_var(
+			$wpdb->prepare(
+				"
+				SELECT COUNT( * ) FROM {$wpdb->prefix}posts
+				WHERE ID=%d
+				AND {$wpdb->prefix}posts.post_content LIKE %s
+				",
+				array( $post_id, $wildcarded )
+			)
+		);
+
+		return ( '0' !== $result ) ? 'Yes' : 'No';
+	}
+
+	/**
+	 * Get info about the cart & checkout pages.
+	 *
+	 * @return array
+	 */
+	public static function get_cart_checkout_info() {
+		global $wpdb;
+
+		$cart_page_id     = wc_get_page_id( 'cart' );
+		$checkout_page_id = wc_get_page_id( 'checkout' );
```

Comment:
Note that this could return -1, which means cart / checkout page is not properly setup. Should we account for this as a different return value to not mess up the stats?

Reply:
That's a nice idea – perhaps we can add that as another data point later (if useful). I haven't closely reviewed the other data in tracker already, we might track whether these pages are set up or onboarding is completed already.